CSharp/BatchExplorer/Service/BatchService.cs

```diff
@@ -201,13 +201,56 @@ public Task<CloudPool> GetPoolAsync(string poolId)
             int targetDedicated,
             TimeSpan? timeout,
             ComputeNodeDeallocationOption? deallocationOption)
-        {
+        {
+            var pool = await this.GetPoolAsync(poolId);
+            if (pool.AutoScaleEnabled.HasValue && pool.AutoScaleEnabled.Value)
+            {
+                await this.Client.PoolOperations.DisableAutoScaleAsync(poolId);
+                if (!await this.WaitForResizingOperationToFinish(poolId))
+                {
+                    throw new Exception("Autoscale disable operation is already in progress. Please try again after sometime to Resize the pool.");

```

Comment:
Prefer a more specific exception type (I know we have a lot of base Exceptions already but would be good to start turning that around)